examples/hpc/dla-driver/src/lib.rs

```diff
@@ -1,11 +1,93 @@
 #![no_std]
 
-mod mmap;
+extern crate alloc;
 
+mod mmap;
+pub use mmap::{
+    DLA0_ADDR, MEMORY_BANK_0_OFFSET, MEMORY_BANK_10_OFFSET, MEMORY_BANK_11_OFFSET,
+    MEMORY_BANK_12_OFFSET, MEMORY_BANK_13_OFFSET, MEMORY_BANK_14_OFFSET, MEMORY_BANK_15_OFFSET,
+    MEMORY_BANK_1_OFFSET, MEMORY_BANK_2_OFFSET, MEMORY_BANK_3_OFFSET, MEMORY_BANK_4_OFFSET,
+    MEMORY_BANK_5_OFFSET, MEMORY_BANK_6_OFFSET, MEMORY_BANK_7_OFFSET, MEMORY_BANK_8_OFFSET,
+    MEMORY_BANK_9_OFFSET, MEMORY_BANK_BASE_ADDR,
+};
+
+use alloc::vec::*;
 use core::ptr;
-use headsail_bsp::sprint;
+use headsail_bsp::{sprint, sprintln};
 use mmap::*;
 
+pub struct Dla {
+    simd_mode: SimdBitMode,
```

Comment:
Is it necessary to cache this value? Can the hardware do it for us? Reduce runtime state if possible. This is an easy way to make hard to track implementation bugs less likely.

Reply:
Unnecessary caching removed and replaced with get calls to hardware.